crates/vim/src/object.rs

```diff
@@ -607,6 +614,99 @@ fn expand_to_include_whitespace(
     range
 }
 
+/// If not `around` (i.e. inner), returns a range that surrounds the paragraph
+/// where `relative_to` is in. If `around`, principally returns the range ending
+/// at the end of the next paragraph.
+///
+/// Here, the "paragraph" is defined as a block of non-blank lines or a block of
+/// blank lines. If the paragraph ends with a trailing newline (i.e. not with
+/// EOF), the returned range ends at the trailing newline of the paragraph (i.e.
+/// the trailing newline is not subject to subsequent operations).
+///
+/// Edge cases:
+/// - If `around` and if the current paragraph is the last paragraph of the
+///   file and is blank, then the selection results in an error.
+/// - If `around` and if the current paragraph is the last paragraph of the
+///   file and is not blank, then the returned range starts at the start of the
+///   previous paragraph, if it exists.
+fn paragraph(
+    map: &DisplaySnapshot,
+    relative_to: DisplayPoint,
+    around: bool,
+) -> Option<Range<DisplayPoint>> {
+    let mut paragraph_start = start_of_paragraph(map, relative_to);
+    let mut paragraph_end = end_of_paragraph(map, relative_to);
+
+    let paragraph_end_row = paragraph_end.row();
+    let paragraph_ends_with_eof = paragraph_end_row == map.max_buffer_row();
```

Comment:
@noritada max_buffer_row is in the wrong co-ordinate system (c.f. #8081).

```suggestion
    let paragraph_ends_with_eof = paragraph_end_row == map.max_point().row();
```

Reply:
Thank you very much! I have incorporated this change.